### PR TITLE
Improve detection of antimeridian crossings

### PR DESCRIPTION
**Pull request by thabbott**
## Changes

Summary: we shouldn't assume that flights never span more than 180 degree longitude when detecting antimeridian crossings--they sometimes do if flight-level winds strongly favor travel in a particular direction.

Before changes:

image

After changes:

image

### Breaking changes

- Flight antimeridian crossings are now detected based on individual flight segments rather than minimum and maximum longitude values. This allows for correct detection of antimeridian crossings for flights that span more than 180 degrees longitude.

### Fixes

- Improve detection of antimeridian crossings for flights that cover over 180 degrees longitude.

## Tests

- [x] QC test passes locally (`make test`)
- [x] CI tests pass

## Reviewer

> @zebengberg

### CODE REVIEW

File: tests/unit/test_flight.py [hunk 1]
```diff
@@ -816,6 +817,113 @@ def test_antimeridian_long_cross(direction: str, cross_anti: bool) -> None:
...
+    fl = Flight(
+        longitude=longitude,
+        latitude=np.zeros(n),
+        altitude=np.full(n, 10000),
+        time=pd.date_range("2022-01-01", "2022-01-01T06", n),
+    )
+    with pytest.raises(ValueError, match="Cannot handle flight that spans"):
+        fl.resample_and_fill("5s")
+
+
```
Comment (zebengberg) [inline tests/unit/test_flight.py:926]: Is it worth having a test for a flight that zippers back and forth across the anti-meridian? Something like

```
longitude = (np.sin(np.linspace(0, 20, 1000))) % 360.0 - 180.0
```

I would expect a flight with this longitude field to make it through `Flight.resample_and_fill` without any errors.

**Review by zebengberg — APPROVED** [review]
Thanks Tristan for cleaning up these overlooked situations!

File: pycontrails/core/flight.py [hunk 2]
```diff
@@ -1262,6 +1240,54 @@ def distance_to_coords(self: Flight, distance: ArrayOrFloat) -> tuple[
...
+        jump21 = s2[:-1] & s1[1:]  # eastward
+        if not np.any(jump12 | jump21):
+            return None
+
+        # separate flight into segments that are east and west of crossings
+        net_westward = np.insert(np.cumsum(jump12.astype(int) - jump21.astype(int)), 0, 0)
+        if net_westward.max() - net_westward.min() > 1:
+            msg = "Cannot handle consecutive antimeridian crossings in the same direction"
+            raise ValueError(msg)
+        east = (net_westward == 0) if net_westward.max() == 1 else (net_westward == -1)
```
[the remark this answers is not in the document]
Comment (thabbott) [reply pycontrails/core/flight.py]: https://github.com/contrailcirrus/pycontrails/pull/222/commits/3ccb734906cd72993e89f52e211b9294bcaca76a

File: tests/unit/test_flight.py (hunk 1 above)
Comment (thabbott) [reply tests/unit/test_flight.py:964]: https://github.com/contrailcirrus/pycontrails/pull/222/commits/2b783e24d8378aeab1e4f117b6d15b493e9bc274

**Pull request merged**